### PR TITLE
ROX-12980: added PR template and onboarding instructions

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

The PR will add the following:

1. A PR template for a more structured way of creating PRs
2. A CONTRIBUTING.md file that onboards new engineers and gives them information about how things work

## Checklist (Definition of Done)
- ~~[ ] Unit and integration tests added~~
- ~~[ ] Added test description under `Test manual`~~
- [x] Documentation added if necessary
- ~~[ ] CI and all relevant tests are passing~~
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- ~~[ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~~

## Test manual

No tests

### CODE REVIEW

**Comment by sachaudh** [comment]
@RedHatInsights/acs-ui-team please take a look at the PR template and CONTRIBUTING.md and let me know if you have any comments, suggestions, or questions. If you haven't really done much on this repo I think you'd be a perfect candidate to let me know if something isn't very clear in the instructions

**Review by dvail — APPROVED** [review]
Thanks Saif, this is very helpful! I'll approve if you wanted to get it merged but would good to get other feedback too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+The workflow files will do the following based on which branch was updated:
+
+1. Run the build command to get the build files located in the build directory
+2. Clone the build repo and copy over the build directory
+3. Add some extra files necessary for RedHatInsights automation to work properly
+4. Commit the new files to the appropriate branch in the build repo
+
+### Branch updates
+
+Check the [deploying](https://github.com/RedHatInsights/acs-ui#deploying) section for information about how build files will be pushed to the [build repo](https://github.com/RedHatInsights/acs-ui-build)
```
[the remark this answers is not in the document]
Comment (sachaudh) [reply CONTRIBUTING.md:28]: Ah yea that is a good point. The format in this file is the correct assumption. When we push to main, the workflows will push the build files to each of those three branches in the build repo. I've made a few changes recently that might simplify things. Now we only need to worry about `qa-beta`. So we would have `main -> qa-beta`. I will update things

**Review by bradr5 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- main -> qa-beta
+- stable -> qa-stable
+- prod-beta -> prod-beta
+- prod-stable -> prod-stable
+
+We will continue to update these deprecated/obsolete branches for now until further notice.
+
+### Build repo webhook
+
+There is a jenkins webhook configured for the [build repo](https://github.com/RedHatInsights/acs-ui-build) that notifies RedHatInsights automation that changes occured for the branches in that repo. Once RedHatInsights automation is notified, the changes will propagate the appropriate environments
```
[the remark this answers is not in the document]
Comment (sachaudh) [reply CONTRIBUTING.md:52]: @karelhala @Hyperkid123 is it possible to do what Dave mentioned here and be able to see the Jenkins Dashboard?

[the remark this answers is not in the document]
Comment (Hyperkid123) [reply CONTRIBUTING.md:44]: pmed you the link the Jenkins cluster

**Pull request merged**